inst/@symfun/mtimes.m

```diff
@@ -58,7 +58,8 @@
 
 function h = mtimes(f, g)
   [vars, s1, s2] = helper_symfun_binops(f, g);
-  h = symfun(s1 * s2, vars);
+  res = s1 * s2;
+  h = symfun(res, symvar(res));
```

Comment:
> But anyway this has little to do with this PR right?

Well, partly yes. But some tests are failing because they are not getting correct vars in result. Such tests are very less in number so I can leave them now and fix them in next PR.

> Hmm, but this may not be the correct result. Consider

Well, in the example you gave, the previous approach would work because both are `symfun` and both have the same variables.  

> Not sure what you mean

If we consider a symfun `f(x) = x^2` and a sym `y` then `f(x)*y` should have both `x` and `y` in its `vars`. But since the code of `helper_symfun_binops` considers only the variables of the symfun involved in such cases,the result will have only `x` and not `y` included in the `vars`.  
  
I see that my current approach of `symvar` is wrong too. I can change it to take union of the variables from both the operands involved. I think that would definitely work.